lib/vanagon/component.rb

```diff
@@ -231,16 +233,25 @@ def get_build_dir
     # Fetches secondary sources for the component. These are just dumped into the workdir currently.
     #
     # @param workdir [String] working directory to put the source into
-    def get_sources(workdir)
+    def get_sources(workdir) # rubocop:disable Metrics/AbcSize
       sources.each do |source|
         src = Vanagon::Component::Source.source(
           source.url, workdir: workdir, ref: source.ref, sum: source.sum
         )
         src.fetch
         src.verify
+        # set src.file to only be populated with the basename instead of entire file path
+        src.file = File.basename(src.url)
+        extract_with << src.extract(platform.tar) if src.respond_to? :extract
       end
```

Comment:
It just seems like one of those things where if we don't do it now, it'll never happen.. 
I may need more info/context as to what each of these methods are for, but in my mind, when I see a method `do_x` and another method `do_xs`, I expect the latter to be something along the lines of `for xs.each { |x| do_x }`. ya know?